File: macos/Chronos-macOS/TrafficLightManager.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(TrafficLightManager, NSObject)

RCT_EXTERN_METHOD(closeWindow)
RCT_EXTERN_METHOD(minimizeWindow)
RCT_EXTERN_METHOD(toggleFullscreen)

@end

File: macos/Chronos-macOS/TrafficLightManager.swift
import Foundation
import React

@objc(TrafficLightManager)
class TrafficLightManager: NSObject {

  @objc
  func closeWindow() {
    DispatchQueue.main.async {
      if let window = NSApp.mainWindow {
        window.close()
      }
    }
  }
  
  @objc
  func minimizeWindow() {
    DispatchQueue.main.async {
      if let window = NSApp.mainWindow {
        window.miniaturize(nil)
      }
    }
  }
  
  @objc 
  func toggleFullscreen() {
    DispatchQueue.main.async {
      if let window = NSApp.mainWindow {
        window.toggleFullScreen(nil)
      }
    }
  }
  
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return true
  }
}
